### PR TITLE
[mlir][pybind] Support loading DenseElementsAttr of bools

### PR DESCRIPTION
**Pull request by rsuderman**
We were missing the python binding support for loading tensors of booleans from numpy arrays.

### CODE REVIEW

File: mlir/lib/Bindings/Python/IRAttributes.cpp [hunk 1]
```diff
@@ -694,6 +694,8 @@ class PyDenseElementsAttribute
         // f16
         assert(view.itemsize == 2 && "mismatched array itemsize");
         bulkLoadElementType = mlirF16TypeGet(context);
+      // } else if (format == "?") {
```
Comment (stellaraccident) [inline mlir/lib/Bindings/Python/IRAttributes.cpp:697]: Did you possibly push an incorrect patch? This appears to just be commented out

Comment (rsuderman) [reply mlir/lib/Bindings/Python/IRAttributes.cpp:697]: Woops, yes. For this to work we technically need an upstream LLVM change to land. I was uncommenting to test.

**Comment by stellaraccident** [comment]
Speaking of tests, can you add one? There's a unit test for these things I believe.

**Review by stellaraccident — APPROVED** [review]
Thanks. A lint bot says you can format your code better. Once addressed, good to land.

**Comment by rsuderman** [comment]
Looks like there is support just in a slightly different way (packbits required for `i1`

**Pull request closed without merging**